modules/data_warehouse/main.tf

```diff
@@ -55,17 +56,62 @@ resource "random_id" "id" {
   byte_length = 4
 }
 
-# Set up service account for the Cloud Function to execute as
+# Set up Functions service account for the Cloud Function to execute as
+# # Set up the Functions service account
 resource "google_service_account" "cloud_function_service_account" {
   project      = module.project-services.project_id
   account_id   = "cloud-function-sa-${random_id.id.hex}"
   display_name = "Service Account for Cloud Function Execution"
 }
 
-# TODO: scope this down
-resource "google_project_iam_member" "cloud_function_service_account_editor_role" {
+# # Grant the Functions service account Object Create / Delete access
+resource "google_project_iam_member" "cloud_function_service_account_storage_role" {
   project = module.project-services.project_id
-  role    = "roles/editor"
+  role    = "roles/storage.objectAdmin"
+  member  = "serviceAccount:${google_service_account.cloud_function_service_account.email}"
+
+  depends_on = [
+    google_service_account.cloud_function_service_account
+  ]
```

Comment:
nit: explicit dependency is not necessary as email is a computed attrib which make this as implicit dependency